feat(drawer): change drawer layout to dropdown ui

wmontwe opened the pull request:
Resolves #9189

Depends on #9358

This changes the drawer ui to a true dropdown ui solution, hidden behind a feature flag.

## Main Features

1. **Unified Folder Support**: 
   - Added a unified account view that aggregates content from multiple accounts
   - Implemented a unified inbox that shows messages from all accounts in one place

2. **Drawer Layout Improvements**:
   - Added support for two drawer UI modes: dropdown drawer and side rail drawer to keep changes minimal when the feature flag is not enabled
   - Improved folder selection and navigation between accounts
   - Enhanced the account list view with better selection states

3. **UI/UX Enhancements**:
   - Fixed selection state for nested folders
   - Improved avatar color selection for dark mode

## Technical Changes

1. **Architecture**:
   - Created a dedicated repository for unified folders (`UnifiedFolderRepository`)
   - Implemented search functionality to support unified inbox views
   - Added proper message counting for unified folders
   - Created new domain entities: `UnifiedDisplayAccount`, `UnifiedDisplayFolder`, and `UnifiedDisplayFolderType`

2. **Bug Fixes**:
   - Fixed issues with folder selection in nested folders
   - Fixed account selection to only trigger when the account is different
   - Removed deselect functionality that was preventing folders from being selected
   - Fixed management of folders when unified account is shown

3. **Feature Flags**:
   - Added feature flags to control the drawer functionality:
     - `enable_dropdown_drawer`: Controls the overall drawer feature (dropdown vs. side rail)
     - `enable_dropdown_drawer_ui`: Controls the UI mode (dropdown with expandable folders vs. side rail with expandable folders)
     
| Name       | Screenshot                                  |
|-----------------------|----------------------------------------------------|
| Unified Account          | <img src="https://github.com/user-attachments/assets/33cd82fa-ef7a-464e-bae1-7242f11b5e59" alt="Unified Account" width="300" /> |
| Account List          | <img src="https://github.com/user-attachments/assets/5adc9c2b-7600-40c4-a732-8f4bc09789d1" alt="Account List" width="300" /> |
| Account View         | <img src="https://github.com/user-attachments/assets/8d538380-252b-4a3d-a1fa-da0fff41394e" alt="Account View" width="300" /> |
| Account View with nested selection         | <img src="https://github.com/user-attachments/assets/65c7b47b-4299-4b4b-bc97-a6f2369e95af" alt="Account View" width="300" /> |

rafaeltonholo commented on feature/navigation/drawer/dropdown/src/main/kotlin/net/thunderbird/feature/navigation/drawer/dropdown/ui/setting/AccountSettingList.kt:
```diff
@@ -0,0 +1,29 @@
+            .padding(vertical = MainTheme.spacings.default)
+            .fillMaxWidth(),
+    ) {
+        SettingListItem(
+            label = stringResource(id = R.string.navigation_drawer_dropdown_action_sync_all_accounts),
+            onClick = onSyncAllAccountsClick,
+            icon = Icons.Outlined.Sync,
+        )
```
nit: I would consider adding another option: "Add new account"

wmontwe replied:
Added 👍

rafaeltonholo commented on feature/navigation/drawer/dropdown/src/main/kotlin/net/thunderbird/feature/navigation/drawer/siderail/ui/account/SideRailAccountIndicator.kt:
```diff
@@ -18,7 +21,7 @@ internal fun AccountIndicator(
                 minHeight = MainTheme.spacings.default,
             ),
-        color = calculateAccountColor(accountColor),
+        color = calculateAccountColor(accountColor.toArgb()),
```
I would encapsulate this within a remember function. The `calculateAccountColor` function performs color calculations that it would be good to avoid recalculating whenever a recomposition occurs.

```suggestion
        color = remember(accountColor) {
            calculateAccountColor(accountColor.toArgb())
        },
```

wmontwe replied:
I removed the `toArgb()` calculation. But remember would require more changes, as it's not a Composable context.

wmontwe replied:
I think that works, thanks

rafaeltonholo reviewed (commented):
Added a first round of review. I will come back later to complete the review

wmontwe commented on core/ui/compose/theme2/common/src/main/kotlin/app/k9mail/core/ui/compose/theme2/SelectThemeColorScheme.kt:
```diff
@@ -132,3 +133,9 @@ fun Color.toColorRoles(context: Context): ColorRoles {
     )
 }
+
+@SuppressLint("RestrictedApi")
+fun Color.toSurfaceContainer(context: Context): Color {
+    val surfaceContainer = MaterialColors.getSurfaceContainerFromSeed(context, this.toArgb())
+    return Color(surfaceContainer)
+}
```
I used an alpha value previously, but that caused issues due to the brackground shining through. I'll check out `compositeOver`.

rafaeltonholo commented on feature/navigation/drawer/dropdown/src/main/kotlin/net/thunderbird/feature/navigation/drawer/dropdown/ui/account/AccountListItem.kt:
```diff
@@ -1,30 +1,82 @@
+            AvatarOutlined(
+                color = color,
+                name = name,
+                size = AvatarSize.MEDIUM,
```
I see, thanks for explaining that

rafaeltonholo commented on feature/navigation/drawer/siderail/src/main/kotlin/net/thunderbird/feature/navigation/drawer/siderail/SideRailDrawer.kt:
```diff
@@ -79,6 +79,7 @@ class SideRailDrawer(
     override fun selectUnifiedInbox() {
         drawerState.update {
             it.copy(
+                selectedAccountUuid = "unified_account",
```
As this is being used in multiple places, I believe this could be moved to a constant

rafaeltonholo reviewed (changes requested):
Great work on the new Drawer implementation! I'm excited to see this new version live soon!

I have just left a few comments that need to be addressed before merging. Let me know you have any questions

wmontwe commented on feature/account/avatar/impl/src/main/kotlin/net/thunderbird/feature/account/avatar/ui/AvatarOutlined.kt:
```diff
@@ -51,19 +51,19 @@ fun AvatarOutlined(
 ) {
     Surface(
-        color = colorRoles.accentContainer,
+        color = color,
```
Good catch!

wmontwe commented:
> I would like to give two suggestions that could be included in this PR or in a different PR if you feel it would make sense to add that:
> 
>     1. Add an option to "Add new account" in the Settings options when the account list is visible. That will improve the user experience, giving a fast way to add a new account
> 
>     2. Enable account switching by swiping down the `AccountView` when it is displaying a selected account.
> 
> 
> These features would enhance the user experience for me, but I understand they may not be necessary for everyone. Feel free to disregard them if you think they are not a suitable use case.

I updated the pull-request and added your first suggestion. The 2nd is a good idea, but out of scope for these changes. Maybe as separate task.
